code/modules/mob/living/basic/lavaland/goliath/tentacle.dm

```diff
@@ -0,0 +1,119 @@
+/// A tentacle which grabs you if you don't get away from it
+/obj/effect/goliath_tentacle
+	name = "goliath tentacle"
+	icon = 'icons/mob/simple/lavaland/lavaland_monsters.dmi'
+	icon_state = "goliath_tentacle_spawn"
+	layer = BELOW_MOB_LAYER
+	plane = GAME_PLANE
+	anchored = TRUE
+	/// Timer for our current action stage
+	var/action_timer
+	/// Time in which to grab people
+	var/grapple_time = 10 SECONDS
+	/// Lower bound of damage to inflict
+	var/min_damage = 10
+	/// Upper bound of damage to inflict
+	var/max_damage =  15
+
+/obj/effect/goliath_tentacle/Initialize(mapload)
+	. = ..()
+	if (!isopenturf(loc) || isspaceturf(loc) || isopenspaceturf(loc))
+		return INITIALIZE_HINT_QDEL
+	for (var/obj/effect/goliath_tentacle/tentacle in loc)
+		if (tentacle != src)
+			return INITIALIZE_HINT_QDEL
+	if (ismineralturf(loc))
+		var/turf/closed/mineral/floor = loc
+		floor.gets_drilled()
+	deltimer(action_timer)
+	action_timer = addtimer(CALLBACK(src, PROC_REF(animate_grab)), 0.7 SECONDS, TIMER_STOPPABLE)
+
+/obj/effect/goliath_tentacle/Destroy()
+	deltimer(action_timer)
+	return ..()
+
+/// Change to next icon state and set up grapple
+/obj/effect/goliath_tentacle/proc/animate_grab()
+	icon_state = "goliath_tentacle_wiggle"
+	deltimer(action_timer)
+	addtimer(CALLBACK(src, PROC_REF(grab)), 0.3 SECONDS, TIMER_STOPPABLE)
+
+/// Grab everyone we share space with. If it's nobody, go home.
+/obj/effect/goliath_tentacle/proc/grab()
+	for (var/mob/living/victim in loc)
+		if (victim.stat == DEAD || HAS_TRAIT(victim, TRAIT_TENTACLE_IMMUNE))
+			continue
+		visible_message(span_danger("[src] grabs hold of [victim]!"))
```

Comment:
could this be `victim.balloon_alert("grabbed by tentacle!")` or something

Reply:
While I am mostly balloon-pilled I think that combat alerts like these should probably remain in the chatlog. I could put a balloon alert _alongside_ it if you want.

Reply:
that's fine.